internal/bminventory/inventory.go

```diff
@@ -1133,6 +1133,16 @@ func (b *bareMetalInventory) InstallCluster(ctx context.Context, params installe
 	return installer.NewInstallClusterAccepted().WithPayload(&c.Cluster)
 }
 
+func (b *bareMetalInventory) integrateWithAMSClusterPreInstallation(ctx context.Context, cluster *common.Cluster) error {
```

Comment:
Not sure we gain much by making this a separate method unless you want to test it separately from `InstallClusterInternal`

Reply:
There is no much "logical gains" here, there reason I did it that way is in order to maintain a "single point of integration" so when something needs to be updated it is clear what is the integration point.

Also if you will `git grep integrateWithAMS` you will find other "integration points":
```
$ git grep integrateWithAMS
internal/bminventory/inventory.go:              if err = b.integrateWithAMSClusterRegistration(ctx, &cluster); err != nil {
internal/bminventory/inventory.go:func (b *bareMetalInventory) integrateWithAMSClusterRegistration(ctx context.Context, cluster *common.Cluster) error {
internal/bminventory/inventory.go:func (b *bareMetalInventory) integrateWithAMSClusterDeregistration(ctx context.Context, cluster *common.Cluster) error {
internal/bminventory/inventory.go:              if err = b.integrateWithAMSClusterDeregistration(ctx, cluster); err != nil {
internal/bminventory/inventory.go:func (b *bareMetalInventory) integrateWithAMSClusterPreInstallation(ctx context.Context, cluster *common.Cluster) error {
internal/bminventory/inventory.go:                      if err = b.integrateWithAMSClusterPreInstallation(ctx, cluster); err != nil {
internal/bminventory/inventory.go:              if err = b.integrateWithAMSClusterUpdateName(ctx, cluster, *params.ClusterUpdateParams.Name); err != nil {
internal/bminventory/inventory.go:func (b *bareMetalInventory) integrateWithAMSClusterUpdateName(ctx context.Context, cluster *common.Cluster, newClusterName string) error {
```

Reply:
That's what a public API is for right?
If I was searching for places that ocm accounts management was done I would search for the public methods on that interface, not some arbitrary other method naming scheme.

Anyway, this is just a small issue. It's fine for now.

Reply:
Actually that make cense, I will know better for the next time.
Thanks for the feedback.